azurelinuxagent/common/protocol/wire.py

```diff
@@ -998,6 +1020,28 @@ def check_wire_protocol_version(self):
                      "advised by Fabric.").format(PROTOCOL_VERSION)
             raise ProtocolNotFoundError(error)
 
+    def call_function_with_goal_state_refresh_and_retry(self, func):
```

Comment:
Nice idea using functional pointers, gave me an idea on how to improve my PR :)

Reply:
Thanks, good to know!